src/libs/actions/Report.js

```diff
@@ -604,7 +649,11 @@ function updateReportWithNewAction(
 
         // Use updated lastReadSequenceNumber, value may have been modified by setLocalLastRead
         // Here deletedComments count does not include the new action being added. We can safely assume that newly received action is not deleted.
-        unreadActionCount: newMaxSequenceNumber - (lastReadSequenceNumbers[reportID] || 0) - ReportActions.getDeletedCommentsCount(reportID, lastReadSequenceNumbers[reportID] || 0),
+        // If lastReadSequenceNumbers[reportID] is undefined (fetchAllReports isn't completed yet),
+        // fallback to local unreadActionCount and increment it by one each time a pusher event with a new comment arrives.
```

Comment:
>If lastReadSequenceNumbers[reportID] is undefined (fetchAllReports isn't completed yet),

I don't feel very confident that this statement is true (or will always be true). There could be other reasons why a local last read sequence number is `undefined`. By saying "fetchAllReports isn't completed yet" we could really confuse someone trying to debug another case where it's `undefined` for some entirely different reason.

Reply:
Similarly,

>local unreadActionCount and increment it by one each time a pusher event with a new comment arrives

1. Other things **_besides Pusher_** call `updateReportWithNewAction()` so this comment is not truthful
2. What we are calling a "local unreadActionCount" is a little ambiguous here... it looks like we are...

Taking the new sequence number from a newly created action and subtracting any deleted comments from it when we have previously stored a lastReadSequenceNumber for this reportID. Otherwise, we refer to the last stored report's `unreadActionCount` and increment it. Why would the stored report exist but the lastReadSequenceNumbers not exist? If it depends on "fetchAllReports" then why do we have one but not the other.

Thanks for the work here, but is it a great long term solution? Feels to me like we should think about this one some more.